pkg/dns/dns.go

```diff
@@ -516,35 +514,36 @@ func (kd *KubeDNS) Records(name string, exact bool) (retval []skymsg.Service, er
 		validRecord = true
 		break
 	}
+
 	if validRecord {
 		// There is a local service with valid endpoints, return its CNAME.
-		name := strings.Join(reverseArray(path), ".")
+		name := strings.Join(util.ReverseArray(path), ".")
 		// Ensure that this name that we are returning as a CNAME response is a fully qualified
 		// domain name so that the client's resolver library doesn't have to go through its
 		// search list all over again.
 		if !strings.HasSuffix(name, ".") {
 			name = name + "."
 		}
-		glog.Infof("federation service query: Returning CNAME for local service : %s", name)
+		glog.V(2).Infof("federation service query: Returning CNAME for local service : %s", name)
```

Comment:
Spammy debug lines should be V(4). Maybe kubedns is not thorough with log levels anyway so it might be ok. As long as it doesn't show up in a productio cluster on every normal request.